connection.js

```diff
@@ -401,145 +436,368 @@ class Connection {
+            // simple time is not included in reject messages for maintime : no main time, only period time !
+            let universalMaintime = {
+                MinimumMaximumSentence : "",        // minimum/maximum
+                TimecontrolSentence : "",           // main time - initial time and/or max time, etc..
+                SpeedSentence : "",                 // for blitz , live , and corr
+                RankedUnrankedGamesIs : "",         // +/- ranked/unranked games is
+                TimeNumber : 0,                     // for example 600 (600 seconds)
+                TimeToString : "",                  // for example "10 minutes"  = timespanToDisplayString(config.xxx)
+                IncreaseDecreaseSentence : "",      // , please increase/decrease
+                                                    // main time - MaintimeTimecontrolSentence again
+                EndingSentence : "",                // optionnal, for example in canadian, adds explanation
+                ConnBelowAboveSentence : "",        // for conn_log : below/above
+                ConnSentence : "",                  // for conn_log sentence
+                TimecontrolString : "",             /*"fischer" , "simple", "byoyomi" , "canadian" , "absolute"*/
+                SpeedString : "",                   /* "blitz" , "live" , "corr" */
+            };
+
+            // before starting, general information : 
+            // 0) "none" dont have a period time, so it is rejected before the UHMAEATs
+            // (we have sanity checks just in case)
+            // 1) simple time doesn't have a main time, only a period time, 
+            // so we let it slide from maintime rejects
+            // 2) fischer : doesnt have a minperiods or maxperiods
+            // 3) absolute doesnt have a period time
+            // so we let it slide from periodtime rejects
+            // 4) while using gedit "find and replace", make sure you dont replace
+            // t.max_time to t.min_time ! (it doesnt exist !)
+
+            // before starting, make sures : we refresh values //
+            // now just before TimecontrolString is being tested, we again make sure it has the latest value
+            /*"fischer", "byoyomi", "canadian", "simple", "absolute" */
+            this.TimecontrolString = String(t.time_control);
+            // now just before SpeedString is being tested, we again make sure it has the latest value
+            /* "blitz", "live", "correspondence" */
+            this.SpeedString = String(t.speed);
+
+            // before starting, sanity checks //
+            // sanity check : if not fischer, not byoyomi, not canadian, not simple, not absolute
+            if ((this.TimecontrolString !== "fischer") && (this.TimecontrolString !== "byoyomi") && (this.TimecontrolString !== "canadian") && (this.TimecontrolString !== "simple") && (this.TimecontrolString !== "absolute")) {
```

Comment:
Why do this check in this incredible complicated way? Before we did `if (!config.allowed_timecontrols[t.time_control]) {` and that's pretty much the same check i think, If anything, write this check like the last one `!["fischer", "byoyomi", "canadian", "simple", "absolute"].includes(this.TimecontrolString)`.

Reply:
because i didnt know about that yet lol, but indeed it's smaller :laughing: 

however seeing how big the UHMAEAT is, this check is really needed, and it helped in ebugging already, so we'd rather keep it :+1: